File: chat-server-spring/src/main/java/com/netmarble/chat/domain/repository/ChatRoomRepository.java
package com.netmarble.chat.domain.repository;

import com.netmarble.chat.domain.model.ChatRoom;

import java.util.List;
import java.util.Optional;

/**
 * ChatRoom 도메인 리포지토리 인터페이스
 */
public interface ChatRoomRepository {
    
    ChatRoom save(ChatRoom chatRoom);
    
    Optional<ChatRoom> findById(Long id);
    
    List<ChatRoom> findAllActive();
    
    List<ChatRoom> findByCreatorId(Long creatorId);
    
    void delete(ChatRoom chatRoom);
}

File: server/chat-server/src/domain/repository/chat-room.repository.ts
import { ChatRoom } from '../model/chat-room.js';

export const CHAT_ROOM_REPOSITORY = Symbol('ChatRoomRepository');

export abstract class ChatRoomRepository {
  abstract save(chatRoom: ChatRoom): Promise<ChatRoom>;
  abstract findById(id: number): Promise<ChatRoom | null>;
  abstract findAllActive(): Promise<ChatRoom[]>;
  abstract findByCreatorId(creatorId: number): Promise<ChatRoom[]>;
  abstract delete(chatRoom: ChatRoom): Promise<void>;
}
